### PR TITLE
Replace RxJS

### PR DESCRIPTION
**Pull request by siiky**

### CODE REVIEW

File: lib/brain.js [hunk 1]
```diff
@@ -3074,35 +3075,48 @@ Brain.prototype.saveTx = function saveTx (tx) {
...
+      factor: 1.6608,
+      minTimeout: 2000,
+      onFailedAttempt,
     })
+    .then(tx => {
+      this.saveTx(tx) // No big deal if saving the tx details on disk fails
+        .catch(err => console.log("Error saving transaction to local DB (ignoring):", err))
+      return tx
+    })
+    .catch(e => Promise.reject(timedout ? new Error('timeout') : e))
```
Comment (chaotixkilla) [inline lib/brain.js:3154]: Don't think you need the `Promise.reject` call. It's a catch block, which already returns a Promise

File: lib/brain.js [hunk 2]
```diff
@@ -3074,35 +3075,48 @@ Brain.prototype.saveTx = function saveTx (tx) {
...
+        if (err instanceof E.StaleError || err instanceof E.RatchetError) {
+          console.log(
+            err.name === 'StaleError' ?
+              "Local transaction is outdated, won't retry:" :
+              "Local transaction conflicts with server, won't retry:",
+            err.message
+          )
+          return Promise.reject(new pRetry.AbortError(err))
+        }
+        return Promise.reject(err)
```
Comment (chaotixkilla) [inline lib/brain.js:3136]: Same here

**Review by chaotixkilla — APPROVED** [review]

File: lib/brain.js (hunk 1 above)
Comment (siiky) [reply lib/brain.js:3154]: Yeah, the returned value is casted to a Promise if not a Promise already, but it's resolved, not rejected. So I think it is necessary.

File: lib/brain.js (hunk 2 above)
Comment (siiky) [reply lib/brain.js:3136]: Same here xP

The `pRetry.AbortError` tells `pRetry` not to retry the call in this case.

File: lib/brain.js [hunk 3]
```diff
@@ -3115,6 +3129,10 @@ Brain.prototype.updateTx = function updateTx (updateTx) {
       this.tx = tx
       return tx
     })
+    .catch(e => {
+      console.log(e)
+      return newTx
+    })
```
Comment (RafaelTaranto) [inline lib/brain.js]: I'm unsure about this one. There are six functions that call updateTx and only one of them doesn't have proper error handling, with this we'll be masking the error for everyone else.

Also that one usage is in handleScan, at which point there are no bills in the system, so it's better to just fail gracefully over there.

**Review by RafaelTaranto — CHANGES_REQUESTED** [review]

File: lib/brain.js (hunk 3 above)
Comment (siiky) [reply lib/brain.js]: Makes sense, done.

**Pull request closed without merging**